### PR TITLE
a bit of refactoring and error messages

### PR DESCRIPTION
**Pull request by eishay**

### CODE REVIEW

File: shoebox/app/com/keepit/controllers/ChatController.scala [hunk 1]
```diff
@@ -27,13 +27,14 @@ import org.jivesoftware.smack.packet.Message
...
 import play.api.libs.json.JsValue
+import securesocial.core._
 import securesocial.core.providers.FacebookProvider
 import com.keepit.common.db.ExternalId
 //import scala.collection.immutable.Map
 
 object ChatController extends Controller with SecureSocial with Logging {
 
-  def createConnection = {
+  def createConnection() = {
```
Comment (yonatanm) [inline shoebox/app/com/keepit/controllers/ChatController.scala:37]: is it a style thing ? adding () to a methods definition - or semantically different

File: shoebox/app/com/keepit/controllers/ChatController.scala [hunk 2]
```diff
@@ -48,36 +49,32 @@ object ChatController extends Controller with SecureSocial with Logging {
...
 
-  def chat(externalId: ExternalId[User]) = SecuredAction() { implicit request =>
-    val tpl = request.body.asJson.map { json =>
-        ( (json \ "url").asOpt[String],  
-        (json \ "message").asOpt[String] ) 
-    }
-    
-    val url = tpl.get._1.get
-    val message = tpl.get._2.get
+  def chat(externalId: ExternalId[User]) = SecuredAction(false, tolerantJson) { implicit request =>
```
Comment (yonatanm) [inline shoebox/app/com/keepit/controllers/ChatController.scala:52]: nice - what is that ? where did u find it ?

Comment (eishay) [inline shoebox/app/com/keepit/controllers/ChatController.scala:52]: http://www.playframework.org/documentation/api/2.0.1/java/index.html?play/mvc/BodyParser.TolerantJson.html
I bounced into it when I read the Play code, its not well documented

File: shoebox/app/com/keepit/controllers/ChatController.scala (hunk 1 above)
Comment (eishay) [inline shoebox/app/com/keepit/controllers/ChatController.scala:37]: Yes, when adding () you mean that this method may change state somewhere,
without it then its a getter that did not change a state and may be replace
with a var.

**Pull request merged**